packages/experiments/src/components/Pagination/Pagination.base.tsx

```diff
@@ -0,0 +1,233 @@
+
+export class PaginationBase extends BaseComponent<IPaginationProps> {
+  public static defaultProps: Partial<IPaginationProps> = {
+    selectedPageIndex: 0,
+    format: PaginationFormat.buttons,
+    numberOfPageButton: 5,
+    previousPageIconProps: { iconName: 'CaretSolidLeft' },
+    nextPageIconProps: { iconName: 'CaretSolidRight' },
+    firstPageIconProps: { iconName: 'Previous' },
+    lastPageIconProps: { iconName: 'Next' }
+  };
+
+  private _classNames: { [key in keyof IPaginationStyles]: string };
+
+  constructor(props: IPaginationProps) {
+    super(props);
+  }
+
+  public render(): JSX.Element {
+    const {
+      comboBoxAriaLabel,
+      firstPageAriaLabel,
+      previousPageAriaLabel,
+      nextPageAriaLabel,
+      lastPageAriaLabel,
+      firstPageIconProps,
+      previousPageIconProps,
+      nextPageIconProps,
+      lastPageIconProps,
+      pageAriaLabel,
+      pageCount,
+      selectedPageIndex,
+      format,
+      styles,
+      theme,
+      itemsPerPage,
+      totalItemCount
+    } = this.props;
+
+    this._classNames = getClassNames(styles!, {
+      theme: theme!,
+      format: format
+    });
+
+    const canPrevious = selectedPageIndex! > 0;
+    const canNext = selectedPageIndex! + 1 < pageCount;
+    const canFirst = selectedPageIndex !== 0;
+    const canLast = selectedPageIndex !== pageCount - 1;
+
+    if (format === PaginationFormat.comboBox) {
+      const scaleOptions: IComboBoxOption[] = [];
+
+      for (let i = 0; i < this.props.pageCount; i++) {
+        scaleOptions.push({
+          key: `${i}`,
+          text: `${i + 1}`
+        });
+      }
+
+      return (
+        <div className={this._classNames.root}>
+          <IconButton iconProps={firstPageIconProps} onClick={this.handleFirstPage} disabled={!canFirst} aria-label={firstPageAriaLabel} />
+          <IconButton
+            iconProps={previousPageIconProps}
+            onClick={this.handlePreviousPage}
+            disabled={!canPrevious}
+            aria-label={previousPageAriaLabel}
+          />
+          <ComboBox
+            ariaLabel={comboBoxAriaLabel}
+            selectedKey={`${selectedPageIndex}`}
+            options={scaleOptions}
+            onChanged={this.onComboBoxChanged}
+            styles={{
+              container: this._classNames.comboBox
+            }}
+          />
+          <IconButton iconProps={nextPageIconProps} onClick={this.handleNextPage} disabled={!canNext} aria-label={nextPageAriaLabel} />
+          <IconButton iconProps={lastPageIconProps} onClick={this.handleLastPage} disabled={!canLast} aria-label={lastPageAriaLabel} />
+        </div>
+      );
+    }
+
+    let visibleItemLabel;
+    if (itemsPerPage && totalItemCount) {
+      const leftItemIndex = selectedPageIndex! * itemsPerPage + 1;
+      const rightItemsIndex = Math.min((selectedPageIndex! + 1) * itemsPerPage, totalItemCount);
+      visibleItemLabel = `${leftItemIndex} - ${rightItemsIndex} of ${totalItemCount}`;
```

Comment:
This embeds English text. Probably best to be a function that you can override as a prop?

Reply:
I was thinking to ask someone from you team how to do this. Essentially, this is a string template in the format of {0} - {1} of {2} . Is there any existing component in fabric that take a string template as property for either display or aria label ? If there is any existing best practice I can follow, that would be very helpful 

---
In reply to: [233571920](https://github.com/OfficeDev/office-ui-fabric-react/pull/7100#discussion_r233571920) [](ancestors = 233571920)